frontend/src/components/DataSourceConnectionPanel.vue

```diff
@@ -138,16 +137,14 @@ export default defineComponent({
     },
   },
   setup(props) {
-    const store = useStore();
-
     const state = reactive<LocalState>({
       showPassword: false,
     });
 
     const { t } = useI18n();
 
     const database = computed(() => {
-      return store.getters["database/databaseById"](
+      return useDatabaseStore().getDatabaseById(
```

Comment:
Maybe extracting it from the `computed` will be better. computed maybe cause some side effect